File: website/public/src/util/cookie.js
export function getCookie(name) {
    name += "=";
    let ca = document.cookie.split(';');

    for(let i = 0; i < ca.length; i++) {
        let c = ca[i];

        while(c.charAt(0) === ' ') {
            c = c.substring(1,c.length);
        }

        if(c.indexOf(name) === 0) {
            return c.substring(name.length, c.length);
        }
    }

    return null;
}

export function setCookie(name, value, expires, secure, domain) {
    let time = new Date();
    time.setSeconds(time.getSeconds()+expires);
    let cookie = `${name}=${value}; expires=${time.toUTCString()}; domain=${domain}; path=/; samesite=strict;`;

    if(secure) {
        cookie += " secure;";
    }

    document.cookie = cookie;
}

File: auth/user/cookie.go
package user

import (
	"emviwiki/shared/config"
	"emviwiki/shared/constants"
	"net/http"
	"time"
)

// SetSessionCookie sets the access token cookie.
// The cookie is accessible from JavaScript and valid on subdomains.
func SetSessionCookie(w http.ResponseWriter, value string, expires time.Time) {
	http.SetCookie(w, &http.Cookie{
		Name:     constants.AuthCookieName,
		Value:    value,
		Expires:  expires,
		Secure:   config.Get().Server.HTTP.SecureCookies,
		HttpOnly: false,
		Domain:   config.Get().JWT.CookieDomainName,
		Path:     "/",
		SameSite: http.SameSiteStrictMode,
	})
}
